### ISSUE TITLE
Gatsby Compatibility

### ISSUE THREAD

**Issue by joehink** [katspaugh/wavesurfer.js#1841]
### Problem
I'm working on a new website built with [GatsbyJS](https://www.gatsbyjs.org/). During development, Gatsby and wavesurfer.js were working perfectly together. But when it came time to build the website using the `gatsby build` script, I got the following error:

```
"window" is not available during server side rendering.

See our docs page for more info on this error: https://gatsby.dev/debug-html


  13 | 	else
  14 | 		root["WaveSurfer"] = factory();
> 15 | })(window, function() {
     |  ^
  16 | return /******/ (function(modules) { // webpackBootstrap
  17 | /******/ 	// The module cache
  18 | /******/ 	var installedModules = {};


  WebpackError: ReferenceError: window is not defined
  
  - wavesurfer.js:15 Object../node_modules/wavesurfer.js/dist/wavesurfer.js [as js]
    node_modules/wavesurfer.js/dist/wavesurfer.js:15:2
  
  - soundPlayer.js:1 Module../src/components/soundPlayer.js
    src/components/soundPlayer.js:1:1
  
  - soundCard.js:1 Module../src/components/soundCard.js
    src/components/soundCard.js:1:1
  
  - category.js:1 Module../src/templates/category.js
    src/templates/category.js:1:1
```

And here are the options I used to create my wavesurfer instance:

```
WaveSurfer.create({
            container: '#waveform',
            waveColor: '#ccc',
            progressColor: '#008457',
            cursorColor: '#008457',
            cursorWidth: 2,
            responsive: true,
            normalize: true,
            hideScrollbar: true,
            pixelRatio: 1
 });
```

After googling for a while, I discovered that the underlying issue is that the build process operates inside a node environment and not a browser environment and therefore does not have access to the window object.

More specifically the issue stems from the `pixelRatio` option for reasons I don't fully understand, besides the fact that the default value for `pixelRatio` is `window.devicePixelRatio`.

### Solution
Here's the quick fix I used to get around the issue:
(On line 15 of node_modules/wavesurfer.js/dist/wavesurfer.js)

`})(window = { devicePixelRatio: 1 }, function() {`

So basically, if window is undefined set it equal to an object with a `devicePixelRatio` property equal to 1.

I doubt this is the most elegant answer to the problem but I figured I'd share my solution to bring a little awareness to this Gatsby compatibility issue.

Thank you for this package. It really helped me out with my new project!

**Comment by thijstriemstra** [comment]
thanks for documenting the workaround.

**Comment by NSpehler** [comment]
Would you be willing to merge this if we submitted a PR to enable Gatsby compatibility?

**Comment by thijstriemstra** [comment]
sounds good @NSpehler
